test/integration/local/inbound/save-payment-request.test.js

```diff
@@ -106,6 +107,34 @@ describe('save payment requests', () => {
     expect(paymentRequestRow.length).toBe(1)
   })
 
+  test('should only insert the first payment request based on reference Id', async () => {
```

Comment:
Thinking should we have test cases for these next 2 situations where we search the DB for the assigned UUID? This ensures that we know it is being saved down correctly and given that `referenceId` is the first searched attribute.

Reply:
Agreed.  Added